src/components/SignIn/index.js

```diff
@@ -1,49 +1,128 @@
-import React from "react";
+import React, { Component } from "react";
+import Input from "../../components/UI/Input";
 import "./index.css";
 
-const signIn = () => {
-  return (
-    <div className="sign_in_container">
-      <div className="wrapper">
-        <h4>Sign In</h4>
-        <div className="input-group input-group-sm mb-3">
-          <div className="input-group-prepend">
-            <span className="input-group-text" id="inputGroup-sizing-sm">
-              Username
-            </span>
-          </div>
-          <input
-            type="text"
-            className="form-control"
-            aria-label="Sizing example input"
-            aria-describedby="inputGroup-sizing-sm"
-            placeholder="Введите имя"
-          />
-        </div>
-        <div className="input-group input-group-sm mb-3">
-          <div className="input-group-prepend">
-            <span className="input-group-text" id="inputGroup-sizing-sm">
-              Password
-            </span>
+class SignIn extends Component {
+  state = {
+    signInForm: {
+      name: {
+        label: "Username",
+        elementType: "input",
+        elementConfig: {
+          type: "text",
+          placeholder: "Введите имя",
+        },
+        value: "",
+        validation: {
+          required: true,
+          email: true,
+        },
+        valid: false,
+        touched: false,
+      },
+      password: {
+        label: "Password",
+        elementType: "input",
+        elementConfig: {
+          type: "password",
+          placeholder: "Введите пароль",
+        },
+        value: "",
+        validation: {
+          required: true,
+          minLength: 8,
+          consist: /^(?=.*[a-z])(?=.*\d)/,
+        },
+        valid: false,
+        touched: false,
+      },
+    },
+    formIsValid: false,
+  };
+
+  checkValidity = (value, rules) => {
+    let isValid = true;
+    const reg = /\b[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}\b/i;
+    
+
+
+
+    
+    if (rules.required) {
+      isValid = value.trim("") !== "" && isValid;
+    }
+
+    if (rules.email) {
+      isValid = reg.test(value) && isValid;
+    }
+
+    if (rules.minLength) {
+      isValid = value.length >= 8 && isValid;
+    }
+
+    if (rules.consist) {
+      isValid = rules.consist.test(value) && isValid;
+    }
+
+    return isValid;
+  };
```

Comment:
Solution works as designed.
Here you can check some other ways to add a validation per react-final-form package - when validation is a part of a package and moved to a separate abstraction layer.
https://final-form.org/docs/react-final-form/examples/field-level-validation
No need to re-design your solution, just FYI only.